arkouda/pdarrayclass.py

```diff
@@ -1560,3 +1690,156 @@ def argmaxk(pda : pdarray, k : int) -> pdarray:
+
+    ret = None
+    if isinstance(pda, pdarray):
+        repMsg = generic_msg("register {} {}".\
+                             format(pda.name, user_defined_name))
+        ret = create_pdarray(repMsg)
+    elif isinstance(pda, str):
+        repMsg = generic_msg("register {} {}".\
+                             format(pda, user_defined_name))        
+        ret = create_pdarray(repMsg)
+    else:
+        raise TypeError("pda must be pdarray or str")
+    
+    return ret
+
+@typechecked
+def attach_pdarray(user_defined_name : str) -> pdarray:
+    """
+    Return a pdarray attached to the a registered name in the arkouda 
+    server which was registered using register_pdarray()
+    
+    Parameters
+    ----------
+    user_defined_name : str
+        user defined name which array was registered under
+
+    Returns
+    -------
+    pdarray
+        pdarray which points to pdarray registered with user defined
+        name in the arkouda server
+        
+    Raises
+    ------
+    TypeError
+        Raised if user_defined_name is not a str
+
+    See also
+    --------
+    register_pdarray, unregister_pdarray
+
+    Notes
+    -----
+    Registered names/pdarrays in the server are immune to deletion 
+    until they are unregistered.
+
+    Examples
+    --------
+    >>> a = zeros(100)
+    >>> r_pda = ak.register_pdarray(a, "my_zeros")
+    >>> # potentially disconnect from server and reconnect to server
+    >>> b = ak.attach_pdarray("my_zeros")
+    >>> # ...other work...
+    >>> ak.unregister_pdarray(b)
+    """
+    if not isinstance(user_defined_name, str):
+        raise TypeError("user_defined_name must be a str")
+
+    repMsg = generic_msg("attach {}".format(user_defined_name))
+    return create_pdarray(repMsg)
+
+
+@typechecked
+def unregister_pdarray(pda : Union[str,pdarray]) -> None:
+    """
+    Unregister a pdarray in the arkouda server which was previously 
+    registered using register_pdarray() and/or attahced to using attach_pdarray()
+    
+    Parameters
+    ----------
+    pda : str or pdarray
+        user define name which array was registered under
+
+    Returns
+    -------
+    None
+
+    Raises 
+    ------
+    TypeError
+        Raised if pda is neither a pdarray nor a str
+
+    See also
+    --------
+    register_pdarray, unregister_pdarray
+
+    Notes
+    -----
+    Registered names/pdarrays in the server are immune to deletion until 
+    they are unregistered.
+
+    Examples
+    --------
+    >>> a = zeros(100)
+    >>> r_pda = ak.register_pdarray(a, "my_zeros")
+    >>> # potentially disconnect from server and reconnect to server
+    >>> b = ak.attach_pdarray("my_zeros")
+    >>> # ...other work...
+    >>> ak.unregister_pdarray(b)
+    """
+    if isinstance(pda, pdarray):
+        repMsg = generic_msg("unregister {}".format(pda.name))
+    elif isinstance(pda, str):
+        repMsg = generic_msg("unregister {}".format(pda))
+    else:
```

Comment:
the raise TypeError block is no longer needed now that you have the typechecked annotation